Add logging on all RESTful and WebSocket requests in VS Code (#580)
coder/vscode-coder: CHANGELOG.md

### Changed

- Always enable verbose (`-v`) flag when a log directory is configured (`coder.proxyLogDir`).

> - Replaced SSE paths with a one-way WebSocket and centralized creation on `OneWayWebSocket`, and unified socket handling.

Reviewer: I think this is too implementation-specific to deserve a callout in a user-facing changelog.  Probably users do not particularly care if we use SSE or web sockets.  Since the functionality from the user perspective is identical in this case and there are no actions the user can take, we can just omit the change IMO.

### Added

- Add support for CLI global flag configurations through the `coder.globalFlags` setting.

> - Add logging for all REST and some WebSocket traffic, with REST verbosity configurable via `coder.httpClientLogLevel` (`none`, `basic`, `headers`, `body`).

Reviewer: We mention the web socket logs in a point below so I think mentioning here is redundant. 

```suggestion
- Add logging for all REST traffic. Verbosity is configurable via `coder.httpClientLogLevel` (`none`, `basic`, `headers`, `body`).
```

> - An Axios interceptor that tags each request with a UUID to correlate with its response and measure duration.

Reviewer: I feel like the interceptor is too much of an implementation detail, what do you think about something like:

> Both REST and WebSocket logs will include a UUID for correlating requests/responses and opens/closes.

Possibly we also move it below the following line so the order goes: note about REST logs, note about WebSocket logs, note about both.

Author: I changed it to the following:

```
- Add logging for all REST traffic. Verbosity is configurable via `coder.httpClientLogLevel` (`none`, `basic`, `headers`, `body`).
- Add lifecycle logs for WebSocket creation, errors, and closures.
- Include UUIDs in REST and WebSocket logs to correlate events and measure duration.
```

Reviewer: wunderbar!
